src/tests/unit/common/visual-helper-toggle-config-builder.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { VisualHelperToggleConfig } from '../../../assessments/types/test-step';
+import { ManualTestStatus } from '../../../common/types/manual-test-status';
+import {
+    IAssessmentResultType,
+    IGeneratedAssessmentInstance,
+    ITestStepResult,
+} from '../../../common/types/store-data/iassessment-result-data';
+import { VisualizationType } from '../../../common/types/visualization-type';
+import { DetailsViewActionMessageCreator } from '../../../DetailsView/actions/details-view-action-message-creator';
+import { DictionaryStringTo } from '../../../types/common-types';
+import { BaseDataBuilder } from './base-data-builder';
+
+export class VisualHelperToggleConfigBuilder extends BaseDataBuilder<VisualHelperToggleConfig> {
+    private stepKey = 'assessment-1-step-1';
+    private otherKey = 'assessment-1-step-2';
+    constructor() {
+        super();
+        this.data = {
+            assessmentNavState: {
+                selectedTestStep: this.stepKey,
+                selectedTestType: -1 as VisualizationType,
+            },
+            actionMessageCreator: null,
+            instancesMap: {
+                'assessment-1-step-1': {
+                    html: 'html',
+                    propertyBag: {},
+                    target: ['element2'],
+                } as IGeneratedAssessmentInstance,
+            } as DictionaryStringTo<IGeneratedAssessmentInstance>,
+            isStepEnabled: true,
+            isStepScanned: false,
+        };
+    }
+    public withActionMessageCreator(actionMessageCreator: DetailsViewActionMessageCreator): VisualHelperToggleConfigBuilder {
```

Comment:
We can use  `public with<P extends keyof T>(property: P, value: T[P])` directly from `BaseDataBuilder`; `with` has proper typings. In general, do you feel is better to have specific "with" functions or better to use the generic one?

This "with" function should be on the code since before we introduce the `BaseDataBuilder`, that may be the reason we have this but we don't really need them (from my point of view). Then again, I'm not sure how many references we may have to this functions. If we have too many, I'm fine with not changing this right now.

Reply:
Yeah, I think using the generic one is an improvement, but I don't want to change it as part of this PR; I'd rather keep this one closer to a simple move